test/NLog.Extensions.Hosting.Tests/ExtensionMethodTests.cs

```diff
@@ -35,26 +35,51 @@ public void UseNLog_withConfiguration_WorksWithNLog()
 
             var someParam = new NLogProviderOptions { CaptureMessageProperties = false, CaptureMessageTemplates = false };
             var actual = new HostBuilder().ConfigureHostConfiguration(config => config.AddInMemoryCollection(memoryConfig)).UseNLog(someParam).Build();
-            TestHostingResult(actual);
+            TestHostingResult(actual, true);
         }
 
-        private static void TestHostingResult(IHost host)
+        [Fact]
+        public void AddNLog_withShutdownOnDispose_worksWithNLog()
+        {
+            var someParam = new NLogProviderOptions { ShutdownOnDispose = true };
+            var actual = new HostBuilder().ConfigureLogging(l => l.AddNLog(someParam)).Build();
+            try
+            {
+                TestHostingResult(actual, false);
+                Assert.NotNull(LogManager.Configuration);
+            }
+            finally
+            {
+                actual.Dispose();
+                Assert.Null(LogManager.Configuration);
+            }
+        }
+
+        [Fact]
+        public void UseNLog_withAddNLog_worksWithNLog()
+        {
+            var actual = new HostBuilder().UseNLog().ConfigureServices((h,s) => s.AddLogging(l => l.AddNLog())).Build();
+            TestHostingResult(actual, true);
+        }
+
+        private static void TestHostingResult(IHost host, bool resetConfiguration)
```

Comment:
are these tests prepared for parallel running? See https://github.com/NLog/NLog.Extensions.Logging/issues/326

(the tests here are running in parallel, so we should take care when using logmanager etc)

Reply:
Nope not for parallel. Since using the ugly method `Config.SimpleConfigurator.ConfigureForTargetLogging`

Reply:
And because all the AddNLog() and UseNLog() doesn't get a LogFactory as parameter. Only NLogLoggingProvider can take a LogFactory.

Reply:
Something for a different PR

Reply:
>Something for a different PR

OK, but then we need to decorate the test class with a ` [Collection(TestsWithInternalLog)]` (maybe renaming TestsWithInternalLog?)

Reply:
See https://xunit.net/docs/running-tests-in-parallel.html

Reply:
Again outside the scope of this PR. All tests within the same test-class are not run in parallel. Easy solution is just to disable parallel testing for all tests.

Reply:
I'm afraid not. I think The LogManager in this class clashes with LogManager in another class. There for https://github.com/NLog/NLog.Extensions.Logging/issues/326 (it's only flaky in this PR)

Reply:
You have two flaky test classes that affect each other. This is not changed by adding one extra test to one of the test classes. Again it is another issue. Don't worry I will kill the test parallelism in a new PR. But the clash between test classes is unrelated to this PR,

Reply:
> . Don't worry I will kill the test parallelism in a new PR.

I hope you don't mean "Easy solution is just to disable parallel testing for all test-classes.". I don't like to move to long build here too ;)

Reply:
NLog has 2000 unit-tests. This one has 100 unit-test. Don't think you will notice the difference between sequential and parallel. But I agree when NLog stops being static by nature, then it would be great to have parallel all the way.

Reply:
I started some time ago with changing it in NLog (replace logManager with Logfactory), but it was too much so I started in the smaller repos to get experience)